qfieldsync/gui/package_dialog.py

```diff
@@ -150,24 +159,40 @@ def get_export_filename_suggestion(self) -> str:
         )
         return str(full_project_name_suggestion)
 
-    def show_package_page(self):
-        self.nextButton.setVisible(False)
-        self.button_box.setVisible(True)
-        self.stackedWidget.setCurrentWidget(self.packagePage)
-
-    def run_package_project(self) -> None:
+    def check_path_length_and_extension(self):
+        """Check if the file path exceeds 200 characters and the extension is .qgs."""
         export_packaged_project = Path(self.packagedProjectFileWidget.filePath())
 
-        if export_packaged_project.suffix != ".qgs":
-            QMessageBox.critical(
-                self,
-                self.tr("Invalid Filename"),
-                self.tr('The filename must have a ".qgs" extension.'),
+        if export_packaged_project.suffix.lower() != ".qgs":
+            self._set_feedback(
+                self.tr('The filename must have a ".qgs" extension'), disable_save=True
             )
             return
 
-        else:
-            self.package_project()
+        if len(export_packaged_project.as_posix()) > self.MAX_LENGTH_CHARS_FILEPATH:
+            self._set_feedback(
+                self.tr(
+                    "Warning: File path exceeds 200 characters. "
+                    "Longer paths may not be handled properly by your file system."
+                ),
+                disable_save=True,
+            )
+            return
+
+        self._set_feedback("", disable_save=False)
+
+    def _set_feedback(self, message, disable_save=True):
+        """Set feedback message and enable/disable the Save button."""
+        self.packagedProjectFileFeedbackLabel.setText(message)
+        self.packagedProjectFileFeedbackLabel.setVisible(bool(message))
+        self.button_box.button(QDialogButtonBox.StandardButton.Save).setEnabled(
+            not disable_save
+        )
+
+    def show_package_page(self):
+        self.nextButton.setVisible(False)
+        self.button_box.setVisible(True)
+        self.stackedWidget.setCurrentWidget(self.packagePage)
```

Comment:
Is there a particular reason you moved this down? If yes, it is ok. If not, you can keep your diff shorter by keeping things where they were.

Reply:
Not really, just wanted to set function `check_path_length_and_extension`  after the function `get_export_filename_suggestion` but you are right, is better to avoid complications.